libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -154,6 +154,13 @@ void IRGeneratorForStatements::initializeStateVar(VariableDeclaration const& _va
 	}
 }
 
+void IRGeneratorForStatements::initializeLocalVar(VariableDeclaration const& _varDecl)
+{
+	solAssert(m_context.isLocalVariable(_varDecl), "Must be a local variable.");
+	IRVariable zero = zeroValue(*_varDecl.type());
+	m_code << m_context.localVariable(_varDecl).commaSeparatedList() << " := " << zero.commaSeparatedList() << "\n";
```

Comment:
Nice!

Reply:
This won't work for tuples, though, will it? Or do we have yul tuple assignments now?

I would have expected:
```suggestion
	assign(m_context.localVariable(_varDecl), zero);
```

Reply:
Well, it'll never be tuples, but it could still be a multi-slot type, e.g. an external function pointer, so I think it should use ``assign(...)``.

Reply:
Also this indicates that we have a lack of tests here :-).

Reply:
Yea that's how it is locally right now, haven't pushed yet